### PR TITLE
update: Modernize eprihlaska for 2024

### PR DESCRIPTION
**Pull request by mrshu**
* Track dependencies with Poetry

* Bump the dependencies to 2024

* Get rid of Facebook OAuth option

### CODE REVIEW

**Comment by mrshu** [comment]
@Adman @tvinar probably won't be merged before the end of 2024 season but we ought to be ready for the future, not the past (in which the dependencies are pinned for the archaic Python version 3.7.9).

**Review by Adman — APPROVED** [review]
Looks good!

File: eprihlaska/views.py [hunk 1]
```diff
@@ -802,13 +801,13 @@ def create_or_get_user_and_login(site, token, name, surname, email):
 @app.route('/google/login', methods=['GET'])
 def google_login():
     callback_uri = url_for('google_authorize', _external=True)
-    return google.authorize_redirect(callback_uri)
+    return oatuh.google.authorize_redirect(callback_uri)
```
Comment (TomiBelan) [inline eprihlaska/views.py:804]: Typo!

Comment (mrshu) [reply eprihlaska/views.py:804]: Ugh, thanks @TomiBelan!

**Pull request merged**